airflow/models/taskinstance.py

```diff
@@ -166,6 +166,16 @@ class TaskReturnCode(Enum):
     """When task exits with deferral to trigger."""
 
 
+def get_current_context() -> Context:
+    """Retrieve the execution context dictionary without altering user method's signature."""
+    if not _CURRENT_CONTEXT:
+        raise AirflowException(
+            "Current context was requested but no context was found! "
+            "Are you running within an airflow task?"
+        )
+    return _CURRENT_CONTEXT[-1]
```

Comment:
Why do we need to move this? The import is done locally anyway and from what I can tell it can live where it has been.

Reply:
Currently I duplicated it, but I think we should move it to TI (or base operator class), since it is useful in all the operators and not just in Python operators. WDYT?